ui-v2/app/mixins/creating-route.js

```diff
@@ -0,0 +1,23 @@
+import Mixin from '@ember/object/mixin';
+import { get } from '@ember/object';
+
+/**
+ * Used for create-type Routes
+ *
+ * 'repo' is standardized across the app
+ * 'item' is standardized across the app
+ *  they could be replaced with `getRepo` and `getItem`
+ */
+export default Mixin.create({
+  beforeModel: function() {
+    get(this, 'repo').invalidate();
+  },
+  deactivate: function() {
+    // TODO: This is dependent on ember-changeset
+    // Change changeset to support ember-data props
+    const item = get(this.controller, 'item.data');
+    if (get(item, 'isNew')) {
+      item.destroyRecord();
```

Comment:
I think we went through this back when the consul v2 rebuild first went out. ember-data leaves unsaved objects hanging around, so if you don't unload them empty ones can end up in your lists. I'm pretty sure `rollbackAttributes` just resets the object, it doesn't completely wipe it from ember-data's cache?

This is another bit that I've just used from whats happening in the app previously, I wouldn't want to change it now for no reason and introduce any errors.

Reply:
If the object isNew, ember-data unloads the model from the store when you call `rollbackAttributes` - just brought it up since you were checking isNew already.

Reply:
I've added this to my todo to look at changing 🔜 (right now I use it everywhere else in the app so I don't want to tweak in this PR)